pkg/sidecar/sidecar.go

```diff
@@ -66,20 +73,62 @@ func New(configPath string, log logrus.FieldLogger) (*Sidecar, error) {
 	}, nil
 }
 
+// RunDaemon starts the main loop
+// Starts the workload API client to listen for new SVID updates
+// When a new SVID is received on the updateChan, the SVID certificates
+// are stored in disk and a restart signal is sent to the proxy's process
+func (s *Sidecar) RunDaemon(ctx context.Context) error {
+	var wg sync.WaitGroup
+
+	if s.config.SvidFileName != "" && s.config.SvidKeyFileName != "" && s.config.SvidBundleFileName != "" {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			err := workloadapi.WatchX509Context(ctx, &x509Watcher{sidecar: s}, s.getWorkloadAPIAdress())
+			if err != nil && status.Code(err) != codes.Canceled {
+				s.config.Log.Errorf("Error watching X.509 context: %v", err)
+			}
+		}()
+	}
+
+	if s.config.JWTBundleFilename != "" {
```

Comment:
in configuration all JWT are required, but here we have different routines, this means that maybe some user does not want to have bundles, but only tokens, or token but not bundles...
so why not to allow that to happens?